### ISSUE TITLE
"Connect to WiFi" does not work on Pixel 6

### ISSUE THREAD

**Issue by tor-forsale** [joaomgcd/TaskerSettings#55]
Phone: Google Pixel 6
Android 15
TaskerSettings v1.8.0

Fails for any WiFi network with the following error:
```
09.13.18/LicenseCheckerTasker Checking cached only
09.13.18/LicenseCheckerTasker cache validity left -10854
09.13.18/LicenseCheckerTasker Cached status: Licensed
09.13.18/LicenseCheckerTasker Cached only: Licensed
09.13.18/Ew add wait type EasyAction1 time 2147483647
09.13.18/Ew add wait type EasyAction1 done
09.13.18/E add wait task
09.13.18/E Error: 1
09.13.18/E Couldn't find matching configured network

```

**Comment by tor-forsale** [comment]
Did some research and it looks like this bug affects each and every Google Pixel phone for quite a while. Is there any chance it will be fixed anytime soon?

**Comment by joaomgcd** [comment]
Hi. I have Android 15 on my Pixel 6 Pro and it works normally for me, so it doesn't affect every Pixel, at least not mine 😅 Does it maybe start working if you enable device admin for Tasker?

**Comment by tor-forsale** [comment]
Enabled device admin for both Tasker and TaskerSettings. Same error!
Proof that I'm not alone suffering from the same issue/bug:
https://www.reddit.com/r/tasker/comments/1d3kn48/connect_to_wifi_action_stopped_working_on_android/
https://www.reddit.com/r/tasker/comments/1fdclek/connect_to_wifi_not_working_with_tasker_settings/

**Comment by joaomgcd** [comment]
Yes, I'm aware that more people are having the same issue, but I haven't been able to find what's causing it, since it doesn't happen for me :/ 

Could you try enabling all permissions with [this](https://github.com/joaomgcd/Tasker-Permissions) and see if it still happens?

**Comment by tor-forsale** [comment]
I found this in logcat:
```
12-19 10:58:12.881 I/WifiService( 1633): getConfiguredNetworks not allowed for uid=10257
12-19 10:58:12.882 W/Tasker ( 3388): 10.58.12#b#E: warning: Couldn't find matching configured network

```
Do I still miss any permission?

**Comment by BeHappy2021** [comment]
Just got a Nothing 2 with Android 15 and had the same issue. Solved it today (works for me but YMMV):
- Copied the Tasker folder from from my phone to my PC.
- Uninstalled Tasker and everything associated with it (Tasker Settings, AutoInput etc)
- Installed all the Tasker things again and went through setting up all the permissions for each thing. 
- Copied the Tasker folder back to the phone and imported my projects/profiles/tasks
Everything worked again! On mine it just seemed like I was missing some permission that has changed from Android 13 (which was on my original phone that I transferred everything over from) and unless you do a fresh install, it doesn't ask for whatever the new permission is.

**Comment by GFreak01** [comment]
I've got a Pixel 7 with the same issue, I tried the steps that worked for BeHappy2021, but sadly didn't work for me.

**Comment by tor-forsale** [comment]
> I've got a Pixel 7 with the same issue, I tried the steps that worked for BeHappy2021, but sadly didn't work for me.

Thanks for saving my time! I was pretty skeptical about BeHappy2021 as well. 
Anyway, I hope joaomgcd will take this problem seriously at last, and do something to fix the obvious issue

**Comment by GFreak01** [comment]
Intermittent or seemingly device specific problems can be tough.  Joaomgcd isn't having this issue on their device, so it makes it really hard to diagnose.

I'm more annoyed with Google for taking away this functionality.  They should give me a dev option to allow specific apps to access the APIs.  It's stuff like this that is causing Android to feel a lot more like iOS.

**Comment by tor-forsale** [comment]
Agree, it's not easy, but it seems that many Pixel phones (regardless of the exact version) are affected. And arguably Google is the second largest (after Samsung) Android phones manufacture.

**Comment by BeHappy2021** [comment]
It looks like a very random issue with Android 15. I've come across several issues which weren't there on my old Android 13 phone (Zenfone 8 Flip). Some can be solved, others can't. And the randomness would be providing a big headache for the developers.

**Comment by joaomgcd** [comment]
> > I've got a Pixel 7 with the same issue, I tried the steps that worked for BeHappy2021, but sadly didn't work for me.
> 
> Thanks for saving my time! I was pretty skeptical about BeHappy2021 as well. Anyway, I hope joaomgcd will take this problem seriously at last, and do something to fix the obvious issue

I do take it seriously of course, but I just am not able to find what's wrong. As mentioned, it still works for me... The issue is obvious as you mentioned, but the fix is not unfortunately.

**Comment by gzyguo** [comment]
I have this issue right after the migration of LineageOS 22 from 21, both Pixel 8 and Oneplus 7.
Tasker beta 6.4.10 via Play Store
tasker_permissions_portable.exe Jan 26, 2022
Tasker.Settings.1.8.0 and adb install --bypass-low-target-sdk-block FILENAME.apk
But the Tasker Settings' permissions after only manually check via App Info, they are all granted by default, never via system prompts, I even re toggled those permissions manually.

The LOS22 is somewhat not that complete I think since It just been released some days, It can't even prompt to choose Google Sounds app to choose ringtone via System Settings at this stage.

**Comment by GFreak01** [comment]
FYI, the work around that I've started using is to update the "WiFi Scan Period" tasker preference. 

In my use-case, I already have the WiFi network saved and set to auto-connect.  However I want to get connected to my WiFi sooner when I get home.  Therefore, decreasing the scan period to make my phone notice that my WiFi is in range appears to be a valid work around.

**Comment by gzyguo** [comment]
> FYI, the work around that I've started using is to update the "WiFi Scan Period" tasker preference.
> 
> In my use-case, I already have the WiFi network saved and set to auto-connect. However I want to get connected to my WiFi sooner when I get home. Therefore, decreasing the scan period to make my phone notice that my WiFi is in range appears to be a valid work around.

It's a minor improve for you but for me it's to connect 1 wifi (which have proxy configured) from 2 saved wifi at some certain situations. I'll just wait I think to let either Tasker or Lineage fix this. Could be Lineage this time I guess.

**Comment by tor-forsale** [comment]
FYI, March Android system update fix the issue for me. Thanks
